core-bundle/contao/library/Contao/Pagination.php

```diff
@@ -217,11 +219,13 @@ public function generate($strSeparator=' ')
 			return '';
 		}
 
+		$environment = System::getContainer()->get('contao.framework')->getAdapter(Environment::class);
+
 		$blnQuery = false;
-		list($this->strUrl) = explode('?', Environment::get('requestUri'), 2);
+		list($this->strUrl) = explode('?', $environment->get('requestUri'), 2);
 
 		// Prepare the URL
-		foreach (preg_split('/&(amp;)?/', Environment::get('queryString'), -1, PREG_SPLIT_NO_EMPTY) as $fragment)
+		foreach (preg_split('/&(amp;)?/', $environment->get('queryString'), -1, PREG_SPLIT_NO_EMPTY) as $fragment)
```

Comment:
Same as above, changes are not related to the PR (and not necessary).

Reply:
Same as above.